### PR TITLE
Error out on invalid ingress frontend configuration

### PR DESCRIPTION
**Pull request by skmatti**
Refer: https://github.com/kubernetes/ingress-gce/pull/905#discussion_r348098467

### CODE REVIEW

File: pkg/loadbalancers/l7.go [hunk 1]
```diff
@@ -152,10 +154,12 @@ func (l *L7) UrlMap() *composite.UrlMap {
 }
 
 func (l *L7) edgeHop() error {
-	// Keeps track if we will "try" to setup frontend resources based on user configuration.
-	// If user configuration dictates we do not, then we emit an event.
-	willConfigureFrontend := false
 	sslConfigured := l.runtimeInfo.TLS != nil || l.runtimeInfo.TLSName != ""
+	// Return an error if user configuration species that both HTTP & HTTPS are not to be configured.
+	if !l.runtimeInfo.AllowHTTP && !sslConfigured {
+		l.recorder.Eventf(l.runtimeInfo.Ingress, corev1.EventTypeWarning, "WillNotConfigureFrontend", "Will not configure frontend based on Ingress specification. Please check your usage of the 'kubernetes.io/ingress.allow-http' annotation.")
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: I don't think you need to emit an event here. When you bubble up the error, it should get emitted as an event

File: pkg/loadbalancers/l7.go [hunk 2]
```diff
@@ -47,6 +47,8 @@ const (
 	// DefaultPath is the path used if none is specified. It is a valid path
 	// recognized by GCE.
 	DefaultPath = "/*"
+
+	invalidConfigErrorMessage = "error invalid ingress frontend configuration"
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: remove the word error

File: pkg/loadbalancers/l7.go [hunk 3]
```diff
@@ -196,11 +197,6 @@ func (l *L7) edgeHop() error {
 		}
 		klog.V(2).Infof("Successfully deleted unused HTTPS frontend resources for load-balancer %s", l)
 	}
-
-	if !willConfigureFrontend {
-		l.recorder.Eventf(l.runtimeInfo.Ingress, corev1.EventTypeNormal, "WillNotConfigureFrontend", "Will not configure frontend based on Ingress specification. Please check your usage of the 'kubernetes.io/ingress.allow-http' annotation.")
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: I feel like the error message here is much better than what you have now. Can you try to incorporate some of this in to the new message?

Comment (skmatti) [reply pkg/loadbalancers/l7.go]: Done

**Comment by rramkumar1** [comment]
LGTM

/assign @MrHohn for sanity check.

**Pull request merged**